### PR TITLE
Backport pallet info pub to SDK 1.14.0

### PR DESCRIPTION
**Pull request by franciscoaguirre**
Backporting https://github.com/paritytech/polkadot-sdk/pull/4976 to version 1.14.0.

The fields in `PalletInfo` were not `pub` in XCMv4 when they were for XCMv3.

### CODE REVIEW

File: prdoc/pr_4976.prdoc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Forgot to make v4 PalletInfo fields public. Without them we cannot make use of the struct.
+
+crates:
+  - name: staging-xcm
+    bump: patch
+    validate: false
```
Comment (ggwpez) [inline prdoc/pr_4976.prdoc:14]:
```suggestion
```

I hope that we fixed the semver check in the meantime. Can you maybe try to see if the CI goes green without this line?

Comment (franciscoaguirre) [reply prdoc/pr_4976.prdoc:14]: One question I had is whether or not the prdoc needs to be backported as well

Comment (franciscoaguirre) [reply prdoc/pr_4976.prdoc]: Still updated it though

File: prdoc/pr_4976.prdoc
Comment (seadanda) [inline prdoc/pr_4976.prdoc:1]: filename should match the pr number: `pr_5202.prdoc`

**Comment by bkchr** [comment]
@franciscoaguirre do we still need this?

**Comment by franciscoaguirre** [comment]
> @franciscoaguirre do we still need this?

Yes

**Review by ggwpez — APPROVED** [review]

**Pull request merged**